src/datadog/azext_datadog/generated/_params.py

```diff
@@ -0,0 +1,183 @@
+
+    with self.argument_context('datadog api-key list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('monitor_name', type=str, help='Monitor resource name')
+
+    with self.argument_context('datadog api-key get-default-key') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('monitor_name', type=str, help='Monitor resource name', id_part='name')
+
+    with self.argument_context('datadog api-key set-default-key') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('monitor_name', type=str, help='Monitor resource name', id_part='name')
+        c.argument('created_by', type=str, help='The user that created the API key.')
+        c.argument('name', type=str, help='The name of the API key.')
+        c.argument('key', type=str, help='The value of the API key.')
+        c.argument('created', type=str, help='The time of creation of the API key.')
+
+    with self.argument_context('datadog host list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('monitor_name', type=str, help='Monitor resource name')
+
+    with self.argument_context('datadog linked-resource list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('monitor_name', type=str, help='Monitor resource name')
+
+    with self.argument_context('datadog monitored-resource list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('monitor_name', type=str, help='Monitor resource name')
+
+    with self.argument_context('datadog monitor list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+
+    with self.argument_context('datadog monitor show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('monitor_name', options_list=['--name', '-n', '--monitor-name'], type=str, help='Monitor resource '
+                   'name', id_part='name')
+
+    with self.argument_context('datadog monitor create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('monitor_name', options_list=['--name', '-n', '--monitor-name'], type=str, help='Monitor resource '
+                   'name')
+        c.argument('tags', tags_type)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('identity_type', arg_type=get_enum_type(['SystemAssigned', 'UserAssigned']), help='Identity type')
+        c.argument('provisioning_state', arg_type=get_enum_type(['Accepted', 'Creating', 'Updating', 'Deleting', ''
+                                                                 'Succeeded', 'Failed', 'Canceled', 'Deleted', ''
+                                                                 'NotSpecified']), help='Provisioning state')
```

Comment:
The extra `''` in the end is not necessary. @qiaozha to check formatting.

Reply:
I think this is auto generated. '' is used to end this line. I see this formatting in other places as well.

Reply:
@changlong-liu Could you take a look? Thanks

Reply:
Yes, it's a codegen issue.

Hi @limingu , to fix this PR, would you change code manually for short-term solution? 
I just add an issue in codegen for tracking and fixing it: https://github.com/Azure/autorest.az/issues/591